doc/specs/#5000 - Process Model 2.0/#5000 - Process Model 2.0.md

```diff
@@ -0,0 +1,1253 @@
+make up the leaf nodes of the tree. The receiving window process will then be
+able to use that tree to be able to re-create a similar pane structure in its
+window, and use the GUIDs to be able to connect to the content processes for the
+terminals in that tab.
+
+The terminal buffer never needs to move from one process to another - it always
+stays in just a single content process. The only thing that changes is the
+window process which is rendering the content process's swapchain.
+
+Similar to dragging a tab from one window to another window, we can also detect
+when the tab was dropped somewhere outside the bounds of a tab strip. When that
+happens, we'll create a new WT window, and use the data package from the
+drag-drop to initialize the structure of the window.
+
+![tear-out-tab](tear-out-tab.png)
+
+For our own sanity, we'll want to ensure that tabs cannot be torn out from a
+"Preview" Windows Terminal window and dropped into a "Release" window. I'm not
+positive that the system will prevent that on our behalf, so I think it's
+important to call out that if this _is_ possible, we should expressly disallow
+it.
+
+#### Scenario: Mixed Elevation
+
+With the ability for window processes to connect to other pre-existing content
+processes, we can now also support mixed elevation scenarios as well.
+
+If a user requests a new elevated tab from an otherwise unelevated window, we
+can use UAC to create a new, elevated window process, and "move" all the current
+tabs to that window process, as well as the new elevated client. Now, the window
+process is elevated, preventing it from input injection, but still contains all
+the previously existing tabs. The original window process can now be discarded,
+as the new elevated window process will pretend to be the original window.
+
+![mixed-elevation](mixed-elevation.png)
+
+We would probably want to provide some additional configuration options here as
+well:
+* Users will most likely want to be able to specify that a given profile should
+  always run elevated.
+  - We should also provide some argument to the `NewTerminalArgs` (used by the
+    `new-tab` and `split-pane` actions) to allow a user to elevate an otherwise
+    unelevated profile.
+* We'll probably want to create new content processes in a medium-IL context by
+  default, unless the profile or launch args otherwise indicates launching
+  elevated. So a window process running elevated would still create unelevated
+  profiles by default.
+* Currently, if a user launches the terminal as an administrator, then _all_ of
+  their tabs run elevated. We'll probably want to provide a similar
+  configuration as well. Maybe if a window process is initially launched as
+  elevated (rather than inheriting from an existing session), then it could
+  either:
+  - default all future connections it creates to elevated connections
+  - assume the _first_ tab (or all the terminals created by the startup

```

Comment:
We probably need to add another indicator in the tab if the tab is elevated, to make life easier for people.

Reply:
YES WE DO.

Reply:
(this is now in #8455)